@here/harp-mapview/lib/ThemeLoader.ts

```diff
@@ -460,69 +422,50 @@ export class ThemeLoader {
-                        if (typeof style.extends === "string" && styleIdMap.has(style.extends)) {
-                            // extends the existing style referenced by `style.extends`.
-                            const baseStyleIndex = styleIdMap.get(style.extends)!;
-                            const baseStyle = newStyleSet[baseStyleIndex];
-                            newStyleSet[baseStyleIndex] = { ...baseStyle, ...style } as any;
-                            newStyleSet[baseStyleIndex].extends = undefined;
-                            return;
-                        }
-
-                        if (typeof style.id === "string" && styleIdMap.has(style.id)) {
-                            // overrides the existing style with `id` equals to `style.id`.
-                            const styleIndex = styleIdMap.get(style.id)!;
-                            newStyleSet[styleIndex] = style;
-                            return;
-                        }
-
-                        newStyleSet.push(style);
-                    });
+            const newStyles: Styles = [];
+            const styleIdMap = new Map<string, number>();
+            baseStyles.forEach(style => {
+                if (typeof style.id === "string") {
+                    //multiple identical style.ids are not supported and will fall back to the
+                    //first occurence
+                    if (!styleIdMap.has(style.id)) {
+                        styleIdMap.set(style.id, newStyles.length);
+                    }
+                }
+                newStyles.push(style);
+            });
 
-                    styles[styleSetName] = newStyleSet;
+            themeStyles.forEach(style => {
+                if (typeof style.extends === "string" && styleIdMap.has(style.extends)) {
+                    // extends the existing style referenced by `style.extends`.
+                    const baseStyleIndex = styleIdMap.get(style.extends)!;
+                    const baseStyle = newStyles[baseStyleIndex];
+                    newStyles[baseStyleIndex] = { ...baseStyle, ...style } as any;
+                    newStyles[baseStyleIndex].extends = undefined;
+                    return;
+                }
 
-                    // remove the styleset from the incoming list
-                    incomingStyleSets.splice(index, 1);
-                } else {
-                    // copy the existing style set to `styles`.
-                    styles[styleSetName] = baseTheme.styles![styleSetName];
+                if (typeof style.id === "string" && styleIdMap.has(style.id)) {
+                    // overrides the existing style with `id` equals to `style.id`.
+                    const styleIndex = styleIdMap.get(style.id)!;
+                    // only match if the two rules are from the same styleset
+                    if (newStyles[styleIndex].styleSet === style.styleSet) {
+                        newStyles[styleIndex] = style;
```

Comment:
The link leads to the "extend" rule for styles, this is not the case here. This code covers the merging of two styles from different Themes with the same id.
e.g. 

Theme1: 
`{
      id: "special style for roads",
      technique: "line",
     color: "red"

}`


Theme2: 
`{
      id: "special style for roads",
      technique: "line",
     color: "blue"
}`

Still a good point, what should happen in that case, I assumed, that if a Theme brings a style with the same id, it would override the other one, but an extension of attributes could be another option. I had another look into the documentation, seems both cases are not expected :)

The documentation says for this case: 
" Style Rules from extended files are always pre-pended to Style Rules of a current file" and for the duplicate "id"s that would be created: 
"In case there are multiple Style Rules with the same id the [extends](https://docs.in.here.com/static/34304/359973/html/topics/jss.html#extends) behaviour is undefined."

I ll implement the pure prepending of the styles as of the documentation.

Reply:
Decided that we should determine the behaviour of styles with the same IDs in the scope of style.extends ticket mentioned above.